### ISSUE TITLE
Support MOM6 parameter files?

### ISSUE THREAD

**Issue by aekiss** [marshallward/f90nml#163]
Hi Marshall - just a thought: would there be any appetite (and would it make any sense) to add support for reading/writing MOM6 parameter files? 

While they're not actually namelists, they seem pretty similar in logical structure, so I'm guessing the internal representation could remain the same and translation only needs to take place for input/output.

FYI in nmltab I've [bodged together a hacky solution](https://github.com/aekiss/nmltab/blob/9821093ce97fb9f5e4766688013045bd900d194a/nmltab.py#L41-L70) to allow f90nml to read MOM6 files, basically by wrapping it all in one big dummy group.

**Comment by marshallward** [comment]
Generally things have gone awry when we've tried to extend support to namelist-like solutions.  There are few examples in the Issues.  Usually there is some case that is valid in one format and not the other, and then flags start creeping in to track which file we're talking about, making the horrible hand-parser snake nest even more unreadable, if not impossible to parse.

I think that the MOM6 parameter format is very simple compared to namelists, and probably not worth the bother of pulling in all of f90nml.  I put one together that more or less lays out how one ought to look.  (And probably how f90nml ought to look, if I ever bothered to rewrite the scanner/parser).

https://github.com/marshallward/mom6param

I think maybe we'd be better served by adding one to MOM6, or maybe as a separate project.  Despite the better judgement of people who've walked this path, MOM6 has created yet another data format....

**Comment by aekiss** [comment]
Yes, I can see your point about keeping f90nml focused on one format. Thanks for sharing your code, that looks like an elegant approach.

I'd actually forgotten that @micaeljtoliveira had already written code for handling MOM6 input files that seems to do exactly what I want: https://github.com/COSIMA/om3-utils/blob/main/om3utils/mom6_input.py

This is a 2-way translation layer between the MOM6 format and normal namelists, and uses f90nml under the hood.
